pyasdf/treeutil.py

```diff
@@ -84,13 +132,13 @@ def recurse(tree, seen):
                 val = recurse(val, new_seen)
                 if val is not None:
                     result[key] = val
-            if isinstance(tree, Tagged):
-                result = tag_object(tree.tag, result)
+            if hasattr(tree, '_tag'):
+                result = tag_object(tree._tag, result)
         elif isinstance(tree, (list, tuple)):
             new_seen = seen | set([id(tree)])
             result = tree.__class__([recurse(val, new_seen) for val in tree])
-            if isinstance(tree, Tagged):
-                result = tag_object(tree.tag, result)
+            if hasattr(tree, '_tag'):
+                result = tag_object(tree._tag, result)
```

Comment:
Exactly.  Particularly because all of the types actually used are subclasses of `Tagged` so it has to traverse the `__bases__` tree.  Also, duck typing may be handier in the long run.